From 8919b7406dcdd80e779c94b7c0c8e197356f1ae2 Mon Sep 17 00:00:00 2001
From: Heinrich Schuchardt <xypron.glpk@gmx.de>
Date: Tue, 5 Jan 2021 07:52:48 +0100
Subject: [PATCH 1/1] efi_loader: move efi_(u)intn_t to efi.h

Move efi_intn_t and efi_uintn_t to include/efi.h to allow usage without
efi_api.h

Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 include/efi.h     | 5 +++++
 include/efi_api.h | 2 --
 2 files changed, 5 insertions(+), 2 deletions(-)

diff --git a/include/efi.h b/include/efi.h
index 5695273ce9..503fbf060b 100644
--- a/include/efi.h
+++ b/include/efi.h
@@ -20,6 +20,11 @@
 #include <linux/string.h>
 #include <linux/types.h>
 
+/* Type INTN in UEFI specification */
+#define efi_intn_t ssize_t
+/* Type UINTN in UEFI specification*/
+#define efi_uintn_t size_t
+
 /*
  * EFI on x86_64 uses the Microsoft ABI which is not the default for GCC.
  *
diff --git a/include/efi_api.h b/include/efi_api.h
index ecb43a0607..e6e321c4e8 100644
--- a/include/efi_api.h
+++ b/include/efi_api.h
@@ -34,8 +34,6 @@ enum efi_timer_delay {
 	EFI_TIMER_RELATIVE = 2
 };
 
-#define efi_intn_t ssize_t
-#define efi_uintn_t size_t
 typedef void *efi_hii_handle_t;
 typedef u16 *efi_string_t;
 typedef u16 efi_string_id_t;
-- 
2.29.2

